### PR TITLE
feat: add options to remote function

### PR DESCRIPTION
**Pull request by xiaomingplus**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[X] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Hi, I want to use proxy when I called the remote function, but there are no options can pass, so I just add a param at the third param.  So now we can call by this:

```js
parseTorrent.remote(torrentId, (err, parsedTorrent) => {
  if (err) throw err
  console.log(parsedTorrent)
},{
  agent: xxx
})
```

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function parseTorrent (torrentId) {
   }
 }
 
-function parseTorrentRemote (torrentId, cb) {
+function parseTorrentRemote (torrentId, cb, options) {
```
Comment (feross) [inline index.js]: The callback argument should always be last. This is Node.js convention. You need to support it being called without options too, for convenience.

File: index.js [hunk 2]
```diff
@@ -69,11 +69,12 @@ function parseTorrentRemote (torrentId, cb) {
     })
   } else if (typeof get === 'function' && /^https?:/.test(torrentId)) {
     // http, or https url to torrent file
-    get.concat({
+    options = Object.assign({
```
Comment (feross) [inline index.js]: Please call this `opts` instead of `options`

Comment (xiaomingplus) [reply index.js]: ok, I have corrected it.

Comment (feross) [reply index.js]: Thanks! Can you also address my other comment?

**Review by feross — APPROVED** [review]
LGTM! Thanks for the PR @xiaomingplus! Thanks @alxhotel for fixing the merge conflicts, param order, and updating the docs!

**Comment by feross** [comment]
Released as 9.1.0

**Pull request merged**